CMakeLists.txt

```diff
@@ -173,6 +173,10 @@ if(USE_CUDA)
         list(APPEND CUDA_ARCHS "7.5")
     endif()
     list(POP_BACK CUDA_ARCHS CUDA_LAST_SUPPORTED_ARCH)
+    string(SUBSTRING ${CUDA_LAST_SUPPORTED_ARCH} 0 1 MAJOR_VERSION)
+    string(SUBSTRING ${CUDA_LAST_SUPPORTED_ARCH} 2 1 MINOR_VERSION)
+    string(CONCAT CUDA_ARCH_VERSION ${MAJOR_VERSION} ${MINOR_VERSION})
```

Comment:
I guess this will fail once CUDA ARCH become 2-digit.
Maybe we can simply replace `.` with empty string?
Or use some workarounds for splitting a sting: https://gitlab.kitware.com/cmake/cmake/-/issues/21970#note_952813.

Reply:
Could you please explain why do we set only the latest CUDA architecture while XGBoost sets all supported archs?
https://github.com/dmlc/xgboost/blob/e41619b1fc6152d4392896c6caf9a0dc960c5558/cmake/Utils.cmake#L104-L122
https://github.com/dmlc/xgboost/blob/e41619b1fc6152d4392896c6caf9a0dc960c5558/cmake/Utils.cmake#L142-L144

Reply:
> Maybe we can simply replace `.` with empty string?
Good idea!

Reply:
> Sets all supported archs could make compiled code run at all archs of GPUs no matter old ones (Kepler, Maxwell) or new ones (Turing, Ampere)

Thanks a lot of the explanation!